waiter/src/waiter/util/http_utils.clj

```diff
@@ -125,11 +125,11 @@
   (let [http2-transport (prepare-http2-transport conn-timeout socket-timeout)]
     {:http1-client (http-client-factory (cond-> config
                                           client-name (update :client-name str "-http1")
-                                          user-agent (update :user-agent str ".http1")))
+                                          (not (str/blank? user-agent)) (update :user-agent str ".http1")))
      ;; prepare-http2-transport already handles the connect and socket timeouts
      :http2-client (-> (cond-> (dissoc config :conn-timeout :socket-timeout)
                          client-name (update :client-name str "-http2")
-                         user-agent (update :user-agent str ".http2"))
+                         (not (str/blank? user-agent)) (update :user-agent str ".http2"))
```

Comment:
It looks like this is always blank:
https://github.com/twosigma/waiter/blob/e96bcdde1622aa153233a5d6b6410b0995490464/waiter/src/waiter/core.clj#L660-L666

Should we set it to something more useful?

Reply:
There are other code paths where the user-agent isn't blank, .e.g in the tests.